code/game/objects/items/stacks/rods.dm

```diff
@@ -60,30 +67,46 @@ GLOBAL_LIST_INIT(rod_recipes, list ( \
 	else
 		icon_state = "rods"
 
-/obj/item/stack/rods/attackby(obj/item/W, mob/user, params)
-	if(W.tool_behaviour == TOOL_WELDER)
-		if(get_amount() < 2)
-			to_chat(user, span_warning("You need at least two rods to do this!"))
-			return
+/obj/item/stack/rods/welder_act(mob/living/user, obj/item/tool)
+	if(get_amount() < 2)
+		balloon_alert(user, "not enough rods!")
+		return
+	if(tool.use_tool(src, user, delay = 0, volume = 40))
+		var/obj/item/stack/sheet/iron/new_item = new(user.loc)
+		balloon_alert(user, "crafting sheets...")
```

Comment:
there's no need for a balloon alert, we can already tell the activity was a success from the visible message and the rods now being in your hands, since this has no delay.